downstreamadapter/dispatchermanager/helper.go

```diff
@@ -425,7 +517,11 @@ func (h *MergeDispatcherRequestHandler) Handle(eventDispatcherManager *EventDisp
 	for _, id := range mergeDispatcherRequest.DispatcherIDs {
 		dispatcherIDs = append(dispatcherIDs, common.NewDispatcherIDFromPB(id))
 	}
-	eventDispatcherManager.MergeDispatcher(dispatcherIDs, common.NewDispatcherIDFromPB(mergeDispatcherRequest.MergedDispatcherID))
+	if mergeDispatcherRequest.Redo {
```

Comment:
We could add a MergeDispatcher(xxxx, isRedo) function in eventDispatcherManager, to make the code here more clean.